src/pages/SettingsPage.js

```diff
@@ -48,44 +56,154 @@ class Settings extends Component {
     }
   }
 
-  handleSubmit = () => {
-    const { togglKey, workspace, user_ids, currentRate, preference, email } = this.state
-    if (togglKey.length && workspace.length && user_ids.length && currentRate.length && email.length) {
+  generateToken() {
+    const { hubstaffAppToken, hubstaffEmail, hubstaffPass} = this.state
+    axios({
+      method: 'post',
+      url: 'https://api.hubstaff.com/v1/auth',
+      headers: {
+        'App-Token': hubstaffAppToken
+      },
+      data: {
+        email: hubstaffEmail,
+        password: hubstaffPass
+      }
+    })
+    .then((res) => {
+      this.setState({hubstaffAuthToken: res.data.user.auth_token, isModalShown: false})
+      toastr.success('Token created!')
+    })
+    .catch((err) => console.log(err))    
+  }
+
+  handleSettSubmit = () => {
+    const { preference, currentRate } = this.state
+    if (currentRate.length) {
+      this.setState({error: ''});
+      this.settingsSettCustomer({currentRate, preference});
+    } else {
+      toastr.error('Please enter correct details');
+    }
+  }
+  settingsSettCustomer(params) {
+    const { preference, currentRate } = params
+    const userUid = this.props.user.uid
+    firebase.database().ref('users/' + userUid + '/settings').update({
+      currentRate,
+      preference
+    })
+    .then(() => toastr.success('Your settings info saved!'))
+  }
+
+  handleHubSubmit = () => {
+    const { hubstaffAppToken, hubstaffAuthToken } = this.state
+    if (hubstaffAppToken.length && hubstaffAuthToken.length) {
       this.setState({error: ''});
-      this.settingsCustomer({togglKey, workspace, user_ids, email, currentRate, preference});
+      this.settingsHubCustomer({hubstaffAppToken, hubstaffAuthToken});
     } else {
       toastr.error('Please enter correct details');
     }
   }
+  settingsHubCustomer(params) {
```

Comment:
do we really need this function?
do you find handleHubSubmit method to be complicated enough so that we need to move this function outside handleHubSubmit?